csharp/extractor/Semmle.Extraction/FilePattern.cs

```diff
@@ -76,8 +76,20 @@ private static StringBuilder BuildRegex(string pattern)
                             throw new InvalidFilePatternException(pattern, "'**' preceeded by non-`/` character.");
                         if (HasCharAt(i + 2, c => c != '/'))
                             throw new InvalidFilePatternException(pattern, "'**' succeeded by non-`/` character");
-                        sb.Append(".*");
-                        i += 2;
+
+                        if (i + 2 < pattern.Length)
+                        {
+                            // Processing .../**/...
+                            sb.Append("(.*/|)");
+                            i += 3;
+                        }
+                        else
```

Comment:
Do we really need this case?
What about
```csharp
if (i + 2 < pattern.Length)
{
    // Processing .../**/...
    sb.Append("(.*/|)");
}
i += 3;
```
If adding three to the index puts the pointer at the end of the pattern we parsing then we break out of the loop and append `.*`

Reply:
I think this would also work, but I slightly prefer to handle this case explicitly. I think it's going to be easier to understand  and maintain better going forward.